### ISSUE TITLE
Intersection type cannot be assigned to typevar

### ISSUE THREAD

**Issue by brandon-leapyear** [microsoft/pyright#3935]
**Describe the bug**
In a function `T -> T`, having more than one type narrowing conditional fails typechecking.

**To Reproduce**
See below snippet.

**Expected behavior**
It should typecheck.

**Screenshots or Code**
```python
from typing import *
T = TypeVar("T")

def foo(x: T) -> T:
    if not isinstance(x, int):
        raise ValueError()
    if not isinstance(x, str):
        raise ValueError()
    return x
```

**VS Code extension or command-line**
Command line tool, 1.1.270

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.271, which I just published. It will also be included in a future release of pylance.
